ludwig/config_validation/checks.py

```diff
@@ -606,3 +607,38 @@ def check_qlora_requirements(config: "ModelConfig") -> None:  # noqa: F821
 
     if config.quantization and (not config.adapter or config.adapter.type != "lora"):
         raise ConfigValidationError("Fine-tuning and LLM with quantization requires using the 'lora' adapter")
+
+
+@register_config_check
+def check_prompt_task_and_template(config: "ModelConfig") -> None:  # noqa: F821
+    """Checks that prompt's template and task properties are valid, according to the description on the schema."""
+    if config.model_type != MODEL_LLM:
+        return
+
+    # If no prompt is provided, no validation necessary:
+    if not config.prompt:
+        return
+
+    template = config.prompt.template
+    task = config.prompt.task
+    # If no template is provided, task is required:
+    if template and not task:
+        raise ConfigValidationError("A prompt task is required if no template is provided!")
```

Comment:
I don't think the error message here matches the condition. It should be `if not template and not task`, right?

Reply:
yup already fixed